server/src/main/java/com/google/sps/utils/ReportReader.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Author: Kira Toal
+ * Date: June 24, 2020
+ */
+package com.google.sps.utils;
+
+import com.google.sps.utils.Ad;
+import com.google.sps.utils.AdRowProcessor;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+public class ReportReader {
+
+  private static final String CSV_FILE_PATH = "/output.csv";
+  private static final String COLLECTION = "testing"; 
+  private static final int START_ROW_INDEX = 1;  // first csv row that gets read in to DB (1 based)
+  private static final int END_ROW_INDEX = 4; // last csv row (exclusive)
+
+  public static void readCSV(InputStream csvFile) throws Exception {
+    BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(csvFile));      
+    String[] currentRow = {}; // array stores single row of CSV data (one advertisement)
+    // read header rows separately
+    String primaryHeader = bufferedReader.readLine();
+    String secondaryHeader = bufferedReader.readLine();
+    int rowIndex = START_ROW_INDEX; 
+    String row = ""; 
+    while((row = bufferedReader.readLine()) != null && (rowIndex < END_ROW_INDEX)) {
+      /* Regex splits the row into a String array by columns (columns are indicated in 
+      the csv by commas.) It ignores commas inside quotation marks to avoid spliting 
+      individual text fields (the text field "North Carolina, Florida" is not split 
```

Comment:
How does the CSV itself know this?

Reply:
It doesn't. I need the regex to only split on the comma if there's an even number of quotes ahead of it.